plugins/module_utils/karbon/node_pools.py

```diff
@@ -0,0 +1,175 @@
+        if config.get("pool_config"):
+            payload["num_instances"] = config["pool_config"]["num_instances"]
+            payload["ahv_config"]["cpu"] = config["pool_config"]["cpu"]
+            payload["ahv_config"]["memory_mib"] = (
+                config["pool_config"]["memory_gb"] * 1024
+            )
+            payload["ahv_config"]["disk_mib"] = config["pool_config"]["disk_gb"] * 1024
+        subnet_uuid, err = get_subnet_uuid(config["node_subnet"], self.module)
+        if err:
+            return None, err
+        payload["ahv_config"]["network_uuid"] = subnet_uuid
+        if config.get("node_iscsi_subnet"):
+            iscsi_subnet_uuid, err = get_subnet_uuid(
+                config["node_iscsi_subnet"], self.module
+            )
+            if err:
+                return None, err
+            payload["ahv_config"]["iscsi_network_uuid"] = iscsi_subnet_uuid
+
+        return payload, None
+
+    def get_pool_spec(self):
+        default_pool_spec = self._get_default_pool_spec()
+        error = self.validate_pool_resources()
+        if error:
+            return None, error
+        spec, error = self._build_pool_spec(default_pool_spec, self.module.params)
+        if error:
+            return None, error
+        return spec, None
+
+    def get_labels_spec(self):
+        return deepcopy(
+            {
+                "add_labels": self.module.params.get("add_labels"),
+                "remove_labels": self.module.params.get("remove_labels"),
+            }
+        )
+
+    def add_node_pool(self, cluster_name, data=None):
+
+        endpoint = "add-node-pool"
+        resp = self.update(
+            data=data,
+            uuid=cluster_name,
+            method="POST",
+            endpoint=endpoint,
+        )
+        return resp
+
+    def get_nodes_count(self, cluster_name, pool_name):
+        nodes_count = 0
+        pool = self.get_node(cluster_name, pool_name)
+        if pool:
+            nodes_count = len(pool.get("nodes"))
+        return nodes_count
+
+    def remove_nodes_of_pool(self, cluster_name, pool_name):
+        nodes_count = self.get_nodes_count(cluster_name, pool_name)
+        resp = {}
+        if nodes_count:
+            spec = {"count": nodes_count}
+            endpoint = "node-pools/{0}/remove-nodes".format(pool_name)
+            resp = self.update(
+                data=spec, uuid=cluster_name, endpoint=endpoint, method="POST"
+            )
+        return resp
+
+    def remove_node_pool(self, cluster_name, pool_name):
+
+        endpoint = "node-pools/{0}".format(pool_name)
+        resp = self.delete(
+            uuid=cluster_name,
+            endpoint=endpoint,
+        )
+        return resp
+
+    def read_node_pools(self, cluster_name):
+
+        endpoint = "node-pools"
+        resp = self.read(
+            uuid=cluster_name,
+            endpoint=endpoint,
+        )
+        return resp
+
+    def get_node(self, cluster_name, pool_name):
```

Comment:
It should be get_node_pool. as its fetching a particular node pool

Reply:
fixed